tv_apps/smart-home/js/edit.js

```diff
@@ -18,25 +19,50 @@
     editNavElements: undefined,
     currentNode: undefined,
 
-    init: function(spatialNavigator, cardManager, cardScrollable) {
+    init: function(spatialNavigator, cardManager,
+                   cardScrollable, folderScrollable) {
+      var that = this;
+
       this.spatialNavigator = spatialNavigator;
       this.cardManager = cardManager;
       this.cardScrollable = cardScrollable;
+      this.folderScrollable = folderScrollable;
 
       this.regularNavElements =
               [this.searchButton, this.settingsButton, this.editButton];
       this.editNavElements = [this.doneButton, this.addNewFolderButton];
 
       this.cardManager.on('card-swapped', this.onCardSwapped.bind(this));
+      this.cardManager.getCardList().then(function(cardList) {
+        cardList.forEach(function(card) {
+          if (card instanceof Folder) {
+            card.on('card-swapped', that.onCardSwapped.bind(that));
+          }
+        });
+      });
```

Comment:
Well.. you're right. Let's leave it for now. We can discuss delegating by `evt` libs later.